src/test/scala/vsys/blockchain/state/contract/token/ExecuteTokenContractV2InvalidDiffTest.scala

```diff
@@ -0,0 +1,595 @@
+        blockDiffEi.explicitGet().txsDiff.txStatus shouldBe TransactionStatus.ContractInvalidCaller
+      }
+    }
+  }
+
+  property("execute contract white v2 transaction send function failed with sender not in white list") {
+    forAll(preconditionsAndExecuteContractTransactionInvalidWhite) { case (genesis, _, regContract, _,
+    executeContractIssue, _, _, _, invalidSend, _, _, _) =>
+      assertDiffEi(Seq(TestBlock.create(Seq(genesis, regContract, executeContractIssue))),
+        TestBlock.createWithTxStatus(Seq(invalidSend), TransactionStatus.Failed)) { blockDiffEi =>
+        blockDiffEi shouldBe an[Right[_, _]]
+        blockDiffEi.explicitGet().txsDiff.tokenAccountBalance.isEmpty shouldBe true
+        blockDiffEi.explicitGet().txsDiff.txStatus shouldBe TransactionStatus.Failed
+      }
+    }
+  }
+
+  property("execute contract white v2 transaction transfer function failed with sender not in white list") {
+    forAll(preconditionsAndExecuteContractTransactionInvalidWhite) { case (genesis, _, regContract, _,
+    executeContractIssue, _, _, _, _, invalidTransfer, _, _) =>
+      assertDiffEi(Seq(TestBlock.create(Seq(genesis, regContract, executeContractIssue))),
+        TestBlock.createWithTxStatus(Seq(invalidTransfer), TransactionStatus.Failed)) { blockDiffEi =>
+        blockDiffEi shouldBe an[Right[_, _]]
+        blockDiffEi.explicitGet().txsDiff.tokenAccountBalance.isEmpty shouldBe true
+        blockDiffEi.explicitGet().txsDiff.txStatus shouldBe TransactionStatus.Failed
+      }
+    }
+  }
+
+  property("execute contract white v2 transaction deposit function failed with sender not in white list") {
+    forAll(preconditionsAndExecuteContractTransactionInvalidWhite) { case (genesis, _, regContract, _,
+    executeContractIssue, _, _, _, _, _, invalidDeposit, _) =>
+      assertDiffEi(Seq(TestBlock.create(Seq(genesis, regContract, executeContractIssue))),
+        TestBlock.createWithTxStatus(Seq(invalidDeposit), TransactionStatus.Failed)) { blockDiffEi =>
+        blockDiffEi shouldBe an[Right[_, _]]
+        blockDiffEi.explicitGet().txsDiff.tokenAccountBalance.isEmpty shouldBe true
+        blockDiffEi.explicitGet().txsDiff.txStatus shouldBe TransactionStatus.Failed
+      }
+    }
+  }
+
+  property("execute contract white v2 transaction withdraw function failed with sender not in white list") {
+    forAll(preconditionsAndExecuteContractTransactionInvalidWhite) { case (genesis, _, regContract, _,
+    executeContractIssue, _, _, _, _, _, _, invalidWithdraw) =>
+      assertDiffEi(Seq(TestBlock.create(Seq(genesis, regContract, executeContractIssue))),
+        TestBlock.createWithTxStatus(Seq(invalidWithdraw), TransactionStatus.Failed)) { blockDiffEi =>
+        blockDiffEi shouldBe an[Right[_, _]]
+        blockDiffEi.explicitGet().txsDiff.tokenAccountBalance.isEmpty shouldBe true
+        blockDiffEi.explicitGet().txsDiff.txStatus shouldBe TransactionStatus.Failed
+      }
+    }
+  }
```

Comment:
the deposit and withdraw fail check is tricky here since the `contractID` used here can not support deposit and withdraw. @Icermli @twwu123 @Julianxa please double-check this in dev net with some valid contract which supported to deposit token.

Reply:
This works as expected in the Devnet

User in white list deposits into contract -> can withdraw while still in white list

Once updated to remove user from white list, the user cannot withdraw the rest of the funds.